<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<title>Ian J Wilson --  Software Downloads</title>
<style type="text/css">
body {font: 140%; color: #03171C; background: #F1FCFF;}
h1{font: 300%}
h2{font: 200%}
h3{font: 160%}
</style>
</head>
<body>
<h1>Research Page for Ian Wilson</h1>
<p><a href="mailto:ian_j_wilson@outlook.com">ian_j_wilson@outlook.com</a><br>
  <a href = "https://scholar.google.com/citations?user=9m4PKIQAAAAJ&hl=en">Google Scholar Profile</a>
 </p>
  This page contains links to example code, code fragments and source code that are used
  in published and ongoing work.


<hr>
<table><tr><td><a href = "tidy_analysis_by_snp.html">snp summary</td><td><a href = "tidy_analysis_by_gene.html">by gene analysis</td></tr></table>
<hr>
	
<table summary="Available Software"  border=0  cellspacing=5>
  <tbody>
    <tr bgcolor="#D8FFF0">
      <td>September 2023</td>
      <td><a href="https://github.com/ijwilson/SYNOD"><b>SYNOD</b></a> Software in the paper
      <br>
      <em>Copy number analysis from genome sequencing data of 11,754 rare disease parent-child trios: a model for
        identifying autosomal recessive human gene knockouts including a novel gene for autosomal recessive retinopathy.</em>
      E Olinger, IJ Wilson, S Orr, M Barroso-Gil, R Neatu <em> et al.</em> - Genetics in Medicine Open, 2024 <br>
    
   <a href="https://www.sciencedirect.com/science/article/pii/S2949774424009804">link to paper</a></a>
  </td>
  </tr>
    <tr > 
      <td>May 2017</td>
      <td>Diversit-tag.  Code for the paper <em>
      Abundant and equipotent founder cells establish and maintain acute lymphoblastic leukaemia</em> by 
      A Elder, S Bomken, I Wilson <em>et al.</em>. Leukemia volume 31: 2577–2586 is on github at 
      <a href = "https://github.com/ijwilson/diversit-tag">github.com/ijwilson/diversit-tag.</https:></td></tr>
    <tr bgcolor="#D8FFF0">
      <td>January 2017</td><td><a href="http://rpubs.com/ijwilson/242602">JAGS Analysis of Copy Number</a><br />
        A Second document
<a href="http://rpubs.com/ijwilson/244991">Estimating Copy Number Frequencies from Censored Genotype Data</a> looks at the
differences between inference with all genotypes and for censored data.
      </td>
    </tr>
    <tr>
      <td>
        January 2016
      </td>
      <td>
        <a href="http://hmg.oxfordjournals.org/content/25/5/1031">Mitochondrial DNA sequence characteristics modulate the size of the genetic bottleneck</a>  2016.  <em> Wilson et al.</em>
        <br />Hum. Mol. Genet. 25 (5): 1031-1041.<br />
        <a href="https://github.com/ijwilson/mtBottleneck">Code on github</a>
      </td>
    </tr>
    <tr bgcolor="#D8FFF0">
      <td>November 2014</td><td>
        <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1005040">Extreme-depth re-sequencing of mitochondrial DNA finds no evidence of
        paternal transmission in humans.   Pyle <em>et al.</em> 2015.  PloS Genetics. <br />
        <a href="PaternalTransmission/index.html">Data & Code</a></td>
    </tr>
    <tr>
      <td></td>
      <td><a href="http://www.sciencedirect.com/science/article/pii/S0040580907000214">A Markov chain Monte Carlo strategy for sampling from the joint posterior distribution of pedigrees and population
        parameters under a Fisher–Wright model with partial selfing.</a> 2007. Theor. Popul. Biol.
         <br /> 
        The program described in this paper is <em>Inference for Maternal Pedigrees with Partial Selfing <b>IMPPS</b> </em><br />
        <a href="impps/index.html">Home Page</a> </td>
      </tr>
    <tr bgcolor="#D8FFF0">
      <td></td>
      <td><b><a href="QTLtree/index.html">QTLtree</a></b> R library and stand alone program <br />
      <a href="QTLtree/index.html">Home Page</a></td>
    </tr>
    <tr>
      <td>  </td>
      <td>
          <a href="http://jnnp.bmj.com/content/85/3/331.short">
           Titin founder mutation is a common cause of myofibrillar myopathy with early respiratory failure</a>.  2014.  Pfeffer <em> et al.</em><br>
           Code and data <a href="sharedhaplotypes/index.html">Shared Haplotypes</a>.
      </td>
    </tr>
  </tbody>
</table>
<hr>
    
<h2> Public Github repositories</h2>
<p>Simulate genetic correlation induced by assortative mating.</p>
    <h3><a href="https://github.com/ijwilson/gencorram">gencorram</a></h3>
    
    <a href="https://github.com/ijwilson/ShinyPopulationGenetics"><img src="images/first_generations_all.gif"></a>
    
    
    <h3><a href="https://github.com/ijwilson/ShinyPopulationGenetics">Shiny Population Genetics</a></h3>
    
Shiny applications for teaching population genetics

	<a href="https://github.com/ijwilson/ShinyPopulationGenetics"><img src="images/snakedriftscreenshot.png"></a>
<dl>
    <dt>Bottleneck Selection</dt><dd> 	Selection through a bottleneck</dd>
    <DT>SnakeDrift</DT><dd>Simple Pictoral representation of Genetic Drift</dd>
    <dt>SelectionDriftSimulation</dt><dd>	Simulation of Drift and Selection</dd>
    </dl>
 	

    
<h2>R packages</h2>
<p>
  Packages (with varying degree of completeness) for the
  <a href="http://cran.r-project.org/">R</a> software system. Latest versions
  in my <a href="R">R package directory</a><br>
  <br>
  Install packages on unix/linux using the commands<br>
  <code>R CMD INSTALL packagename.tar.gz</code><br>
<p>
  To install on a windows systems use the zip versions
  and install from within R using <br>
  <code>install.packages("packagename.zip",repos=NULL)</code><br>
  <br>
  Where you have downloaded the package to your working directory.  Once installed the
  zip file can be safely deleted.
  <br>
  <br>

<table bgcolor="#C3E3E4" cellspacing="20" summary="R packages">
  <tbody>
    <tr>
      <td><a href="R/genealogy_0.01.tar.gz">genealogy.tar.gz</a></td>
      <td><A href="R/genealogy_0.01.zip">genealogy.zip</A></td>
      <td>A package for the simulation of simple population genetic models</td>
    </tr>
    <tr>
      <td><a href="genomic_1.0.tar.gz">genomic.tar.gz</A></td>
        <td><a href="R/genomic_1.0.zip">genomic.zip</a></td>
        <td>A package for downloading and using HAPMAP and Perlegen
          datasets</td>
        </tr>
        <tr>
          <td><a href="R/pac_0.02.tar.gz">pac.tar.gz</a></td>
          <td><a href="R/pac_0.02.zip">pac.zip</a></td>
          <td>A package using Li &amp; Stephens (2003) model for the
            analysis of genetic data, and my extensions for the analysis of
            data from subdivided populations.</td>
        </tr>
        <tr>
          <td><a href="R/ARG_0.1.tar.gz">ARG.tar.gz</a></td>
          <td><a href="R/ARG_0.1.zip">ARG.zip</a></td>
          <td>Simulate and extract information from Ancestral Recombination
            Graphs.  Requires the <strong>ape</strong> package for plots.</td>
        </tr>
        <tr>
          <td><a href="R/batwing_0.2.tar.gz">batwing.tar.gz</a></td>
          <td><a href="R/batwing_0.2.zip">batwing.zip</a></td>
          <td>Package for postprocessing of BATWING output.</td>
        </tr>
        <tr>
          <td><a href="R/ijwtools_0.1.tar.gz">ijwtools.tar.gz</a></td>
          <td><a href="R/ijwtools_0.1.zip">ijwtools.zip</a></td>
          <td>My own set of utility functions (mainly for dealing with genetic data and the results from my own programs)
            including quick reading from xml files, omnibus tests, Manhattan distances
            between rows in a table, multivariate D' calculation.</td>
          </tr>
        <tr>
          <td><a href="R/snptree_0.1.tar.gz">snptree</a></td>
          <td></td>
          <td>Create and plot lexical trees from SNP data.  Used in paper
            <i>Recent mitochondrial DNA mutations increase the risk of developing multiple common late-onset human diseases</i>,
            by Hudson <i>et al.</i> <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004369">link</a>
          </td>
        </tr>
      </tbody>
    </table>

<p>
  Screenshots of interactive plots from the ARG package. Left plot
  shows a view of an ancestral graph showing genetic variants (bottom
  panel) along with the tree under a single site (top left panel) and
  tree heights and lengths (top right panel). Right plot gives views
  of trees 100 sites apart. Click on either plot for a larger
  view.
</p>
<a href="images/snapshot1.png">
  <img src="images/snapshot1_thumbnail.png" width="800" height="600" border=
    "0" alt=""></a>
<a href="images/snapshot2.png"><img src="images/snapshot2.png" width="256" height="189" border="0" alt=
""></a>
<hr>
<h2>C/C++ Software</h2>
<h3>Installation on Unix Systems</h3>
<p>All Software in this section is in the form of zipped tar files.
  Use<br>
  <code>gunzip -c filename.tar.gz | tar xf -</code> <br>
  then type <code>make</code><br>
  to compile the program. A limited manual for micsat is in the file
  <em>micsat.doc</em>. Other manuals can be downloaded in pdf format.
</p>

<h3>Windows (95/98/NT)</h3>
<p>
  The programs are distributed as executable files in a zip file format
</p>

<h3>Macintosh (MacOS 8.0 or above)</h3>
<p>
  A binhex-ed executable for the Macintosh, MacBatwing, may be downloaded.<br>
  On launching, the program opens a dialog box with a text field. The
  user puts the program arguments (infile outfile seed) into this
  field and hits ok.<br>
  (Compilation on the Mac courtesy of Oliver Pybus - send
  comments/bugs to <a href=
"mailto:Ian.Wilson@ncl.ac.uk">Ian.Wilson@ncl.ac.uk</a>).</p>
<br>
<table border="1" summary="C software">
  <tbody>
    <tr>
      <td width="237">&nbsp;</td>
      <td><b><a href="#micsat">Micsat</a></b></td>
      <td><b><a href="#batwing">Batwing</a></b></td>
      <td><b><a href="#parentage">Parentage</a></b></td>
      <td><b><a href="#hapgen">Hapgen</a></b></td>
      <td><b>Ancestors</b></td>
    </tr>
    <tr>
      <td>Unix</td>
      <td><a href="micsat/micsat.tar.gz">micsat.tar.gz</a></td>
      <td><a href="batwing/batwing.tar.gz">batwing.tar.gz</a></td>
      <td><a href="parentage/parentage.tar.gz">parentage.tar.gz</a></td>
      <td><a href="hapgen/hapgen.tar.gz">hapgen.tar.gz</a></td>
      <td><a href="ancestors.tar.gz">ancestors.tar.gz</a></td>
    </tr>
    <tr>
      <td>Windows 95/NT</td>
      <td><a href="micsat/micsat.zip">micsat.zip</a></td>
      <td><a href="batwing/batwing.exe">batwing.exe</a></td>
      <td><a href="parentage/parentage.exe">parentage.exe</a></td>
      <td>&nbsp;</td>
      <td>&nbsp;</td>
    </tr>
    <tr>
      <td>Macintosh (MacOS 8.0 or above)</td>
      <td>&nbsp;</td>
      <td><a href="batwing/MacBatwing.hqx">MacBatwing</a></td>
      <td>&nbsp;</td>
      <td></td>
      <td></td>
    </tr>
    <tr>
      <td>Manuals (pdf format)</td>
      <td><a href="micsat/micsat.html">micsat</a></td>
      <td><a href="batwing/batguide.pdf">bat_guide.pdf</a></td>
      <td><a href="parentage/parentage.pdf">parentage.pdf</a></td>
      <td></td>
      <td>     </td>
    </tr>
    <tr>
      <td>Examples (inc R code)</td>
      <td>&nbsp;</td>
      <td>&nbsp;</td>
      <td><a href="parentage/parexamples.tar.gz">parexamples.tar.gz</a></td>
      <td>&nbsp;</td>
      <td>&nbsp;</td>
    </tr>
    <tr>
      <td>Postprocessing (S/R code)</td>
      <td>&nbsp;</td>
      <td><a href="batwing/Batwing.R">batwing.r</a></td>
      <td>&nbsp;</td>
      <td>&nbsp;</td>
      <td>&nbsp;</td>
    </tr>
    <tr>
      <td>Web interfaces</td>
      <td>&nbsp;</td>
      <td>&nbsp;</td>
      <td><a href="http://cbsuapps.tc.cornell.edu/parentage.aspx">Web inferface at BioHP.org</a>&nbsp;</td>
      <td>&nbsp;</td>
      <td>&nbsp;</td>
    </tr>
  </tbody>
</table>
<br>
<br>
<hr>
<h2>Program Descriptions</h2>
<table bgcolor="#C3E3E4" cellspacing="30" summary="Micsat description">
  <tr>
    <td><img src="images/micsat1.png" align="bottom" border="0" alt="Micsat logo" WIDTH="180" ></td>
    <td><h3>Micsat</h3>
      <p>This is the program described in the paper "Genealogical
        inference from microsatellite data" by Wilson and Balding published
        in <a href="http://www.genetics.org/content/150/1/499">Genetics</a> 150: 499-510.
      </p>
      <p>A <a href="micsat/micsat.htm">description</a> of the program and
        the downloaded files.
      </p>
    </td>
  </tr>
  <tr>
    <td>
      <h3 align="center"><a name="batwing" id="batwing"><img src=
"images/bat.gif" align="bottom" border="0" width="180" alt="BATWING logo"></a></h3>
    </td>
    <td>
      <p><b>B</b>ayesian <b>A</b>nalysis of <b>T</b>rees <b>W</b>ith <b>I</b>nternal
        <b>N</b>ode <b>G</b>eneration. Batwing is described in
        Wilson, Weale &amp; Balding 2003.Inferences from DNA data:
        population histories, evolutionary processes and forensic match
        probabilities. <i>Journal of the Royal Statistical Society: Series
          A</i>, <b>166</b>: 155-188.<br>
          <a href="batwing/batwing.tar.gz">Download</a></p>
    </td>
  </tr>
<tr>
<td><img src="images/parentage.jpg" align="bottom" border="0" alt="Parentage logo"
height="171" width="180"></td>
<td><font face="Comic Sans MS"><strong>Parentage</strong></font>
<p>Program described in <cite>Emery, Wilson, Craig, Boyle and Noble
(2001). Assignment of paternity groups without access to parental
genotypes: multiple mating and development plasticity in
squid.<a href=
"http://www.blackwell-synergy.com/Journals/member/institutions/issuelist.asp?journal=mec">
Molecular Ecology</a>: 10: 1265-1278.</cite></p>
<a href="parentage/parentage.tar.gz" name="parentage" id="parentage">download</a>
<p>The Computational Biology Service Unit at Cornell University
provides free world-wide access to parentage via a web interface (BioHPC.org).
Using this interface, researchers can run a variety of computationally
intensive bioinformatics and population genetics tools, Parentage being one of them.
<p>
<a href="http://cbsuapps.tc.cornell.edu/parentage.aspx">Web inferface at BioHP.org</a>

</td>
</tr>
</table>
<hr align="center">
<h3>Hapgen</h3>
<p>A program for haplotype reconstruction from genotypes. Work done with <a href="mailto:d.cooper@maths.abdn.ac.uk">David
Cooper</a>.</p>
<p>This program is a Markov chain Monte Carlo method to reconstruct
haplotypes from genotypes. It uses Metropolis coupled chains to
speed up mixing.</p>
<p>Includes options to deal with:</p>
<ul>
<li>Known phase,</li>
<li>Missing Data,</li>
<li>Multiple subpopulations</li>
</ul>
<hr align="center">
<h3>Ancestors</h3>
Inference of migration or population splitting using importance
sampling.
<hr>
<h2><a name="Additional" id="Additional">Additional
Material</a></h2>
<h3><cite>Vickers, McLeod, Spector &amp; Wilson (2001). Assessment of mechanism 
  of skewed X-inactivation by analysis of twins.
<a href="http://www.bloodjournal.org/">BLOOD</a>: 97:1274-1281</cite></h3>
<p>WinBUGS analysis of <a href="bugs/twinsmvn.html">Twins
Multivariate Normal Model</a>, and <a href=
"bugs/twinsbeta.html">Twins Stem Cell Number</a></p>
<hr>
<address><a href=
"mailto:ian_j_wilson@outlook.com">ian_j_wilson@outlook.com</a></address>
<p>Last Modified <i>18 June 2024</i></p>


</body>
</html>